tests/test_end_to_end.py

```diff
@@ -55,11 +58,52 @@ def test_model_info(client):
     ]
 
 
-def test_http(client):
-    response = client.get("/species/{}".format('ECOLX'))
+def test_species(client):
+    response = client.get("/species/ECOLX")
     assert response.status_code == 200
+    assert len(response.json) > 0
+
+
+def test_simulate_wrong_id(client):
+    response = client.post("/models/wrong_id/simulate", json={'message': {}})
+    assert response.status_code == 404
+
+
+def test_simulate_no_message(client):
+    response = client.post("/models/iJO1366/simulate", json={})
+    assert response.status_code == 400
+
+
+def test_simulate_infeasible(client):
+    response = client.post("/models/iJO1366/simulate", json={'message': MESSAGE_FLUXES_INFEASIBLE})
+    result = response.json
+    assert result['fluxes']['ATPM'] == pytest.approx(100)
```

Comment:
Seems to be a reliable test case so far although it seems unreliable to me. Individual flux results are typically unstable because there exist multiple flux patterns for a given objective value. When the solver status is non-optimal then typically all bets are off regarding the flux values.

Reply:
`ATPM` is one of the measured fluxes. So presumably this is testing that when a model is infeasible, if there are measured reactions, then their flux values are set to the mean of the bounds regardless of what was returned by the solver, and that should be reliable.